src/odemis/acq/stream/_sync.py

```diff
@@ -1301,13 +1301,20 @@ def _runAcquisitionEbeam(self, future):
                         # Live update the setting stream with the new data
                         try:
                             self._sccd._onNewData(self._ccd_df, self._acq_data[self._ccd_idx][-1])
+                            if integration_count > 10:
+                                logging.debug("Sum the acquired images one after another")
+                                self._integrateImages(integration_count)
                         except Exception:
                             logging.exception("Failed to update CCD live view")
 
                         n += 1  # number of images acquired so far
 
-                    # integrate/sum images
-                    self._integrateImages(integration_count)
+                    if integration_count <= 10:
+                        # integrate/sum images
+                        logging.debug("Integrate the %s acquired images", n)
+                        self._integrateImages(integration_count)
```

Comment:
Don't do anything special for integ count < 10. You're not really saving time, and the code is twice more complex. In addition, to test the difficult code path, you now have to have at least 10 images.
The only special case "allowed" is integration_count == 1.